fractal_server/app/routes/admin/v2.py

```diff
@@ -307,3 +314,126 @@ async def flag_task_v1_as_v2_compatible(
+    command_parallel: Optional[str] = None
+
+    source: str
+
+    meta_non_parallel: dict[str, Any] = Field(default_factory=dict)
+    meta_parallel: dict[str, Any] = Field(default_factory=dict)
+
+    owner: Optional[str]
+    version: Optional[str]
+
+    docs_info: Optional[str] = None
+    docs_link: Optional[HttpUrl] = None
+
+    input_types: dict[str, bool] = Field(default_factory=dict)
+    output_types: dict[str, bool] = Field(default_factory=dict)
+
+
+class ProjectUser(BaseModel):
+    id: int
+    email: EmailStr
+
+
+class TaskContext(BaseModel):
+
+    task_v2_minimal: TaskV2Minimal
+
+    workflow_id: Optional[int]
+    workflow_name: Optional[str]
+    project_id: Optional[int]
+    project_name: Optional[str]
+
+    project_users: Optional[list[ProjectUser]]
+
+
+@router_admin_v2.get("/task/", response_model=list[TaskContext])
+async def query_tasks(
+    id: Optional[int] = None,
+    source: Optional[str] = None,
+    version: Optional[str] = None,
+    name: Optional[str] = None,
+    owner: Optional[str] = None,
+    only_no_owner: bool = False,
+    max_number_of_results: int = 25,
+    user: User = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+) -> list[TaskContext]:
+
+    if (owner is not None) and (only_no_owner is True):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Cannot query Tasks by owner if `only_no_owner` is True",
+        )
+
+    stm = select(TaskV2)
+
+    if id is not None:
+        stm = stm.where(TaskV2.id == id)
+    if source is not None:
+        stm = stm.where(TaskV2.source == source)
+    if version is not None:
+        stm = stm.where(TaskV2.version == version)
+    if name is not None:
+        stm = stm.where(TaskV2.name.icontains(name))
+
+    if owner is not None:
+        stm = stm.where(TaskV2.owner == owner)
+    elif only_no_owner is True:
+        stm = stm.where(TaskV2.owner == None)  # noqa E711
+
+    res = await db.execute(stm)
+    task_list = res.scalars().all()
+    if len(task_list) > max_number_of_results:
+        await db.close()
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=(
+                f"Too many Tasks ({len(task_list)} > {max_number_of_results})."
+                " Please add more query filters."
+            ),
+        )
+
+    contexts = []
+
+    for task in task_list:
+        stm = (
+            select(WorkflowV2)
+            .join(WorkflowTaskV2)
+            .where(WorkflowTaskV2.workflow_id == WorkflowV2.id)
+            .where(WorkflowTaskV2.task_id == task.id)
+        )
+        res = await db.execute(stm)
+        wf_list = res.scalars().all()
+
+        if wf_list:
+            for workflow in wf_list:
+                contexts.append(
+                    TaskContext(
+                        task_v2_minimal=TaskV2Minimal(**task.model_dump()),
```

Comment:
I think it's best to not include this attribute in `TaskContext`, because it would be repeated N times.
I suggest to make the response something with two attributes, one for what is currently `task_v2_minimal` and one with the list of "contexts".

Reply:
To clarify, I propose a structure like the following:

* `response_model=list[TaskV2Info]`
* `TaskV2Info` has an attribute `task` of type `TaskV2Minimal`
* `TaskV2Info` has an attribute `relationships` of type `list[TaskV2Relationship]`, with a default factory `list`.
* Any item of `TaskV2Info.relationships` (of type `TaskV2Relationship`) has _required_ attributes for workflow/project id/name, and a _required_ attribute `project_users` (which can in principle be empty, although this should never happen).